### PR TITLE
python3Packages.python-hglib: 2.6.2 -> 2.6.2-unstable-2024-06-17

### LINKED ISSUE #311054: Migrate python packages from nose to pynose
The `nose` test runner will never run on Python 3.12, and we're replacing it with `pynose` instead. On top of that, packages relying on nose have largely been unmaintained and might be dead leafs or using another test framework entirely. It is also to be expected that they need to be converted to the PEP517 builder in the process of fixing them.

The following packages are in the `python3Packages` scope and should be addressed first. The final goal is the removal of `nose` and related packages.

- [x] bayespy https://github.com/NixOS/nixpkgs/pull/311489
- [ ] binwalk https://github.com/NixOS/nixpkgs/pull/311156
- [x] biopandas https://github.com/NixOS/nixpkgs/pull/313266
- [ ] blessings https://github.com/NixOS/nixpkgs/pull/311156
- [ ] boto
- [ ] cgroup-utils https://github.com/NixOS/nixpkgs/pull/311156
- [ ] chart-studio https://github.com/NixOS/nixpkgs/pull/311156
- [ ] citeproc-py https://github.com/NixOS/nixpkgs/pull/311156
- [x] cld2-cffi https://github.com/NixOS/nixpkgs/pull/313184
- [x] clint https://github.com/NixOS/nixpkgs/pull/314778
- [x] colormath https://github.com/NixOS/nixpkgs/pull/313342
- [x] cryptacular https://github.com/NixOS/nixpkgs/pull/311201
- [x] cufflinks https://github.com/NixOS/nixpkgs/pull/313159
- [x] django-nose https://github.com/NixOS/nixpkgs/pull/313333
- [x] django-sr https://github.com/NixOS/nixpkgs/pull/314800
- [x] doctest-ignore-unicode https://github.com/NixOS/nixpkgs/pull/313169
- [x] ecos https://github.com/NixOS/nixpkgs/pull/311349
- [ ] flaky https://github.com/NixOS/nixpkgs/pull/312260
- [x] flask-mail https://github.com/NixOS/nixpkgs/pull/314780
- [x] flask-principal https://github.com/NixOS/nixpkgs/pull/317167
- [x] forbiddenfruit https://github.com/NixOS/nixpkgs/pull/317170
- [x] gssapi https://github.com/NixOS/nixpkgs/pull/311072
- [x] globre
- [x] hdfs https://github.com/NixOS/nixpkgs/pull/317173
- [x] hdmedians https://github.com/NixOS/nixpkgs/pull/317195
- [x] hid https://github.com/NixOS/nixpkgs/pull/311056
- [x] hkdf https://github.com/NixOS/nixpkgs/pull/317197
- [x] htmllaundry https://github.com/NixOS/nixpkgs/pull/317200
- [x] iodata https://github.com/NixOS/nixpkgs/pull/324723
- [x] ipdbplugin https://github.com/NixOS/nixpkgs/pull/313263
- [x] iptools https://github.com/NixOS/nixpkgs/pull/317166
- [ ] ipy https://github.com/NixOS/nixpkgs/pull/324725
- [x] libvirt https://github.com/NixOS/nixpkgs/pull/313709
- [x] lockfile https://github.com/NixOS/nixpkgs/pull/311056
- [ ] mhcflurry
- [ ] mkl-service https://github.com/NixOS/nixpkgs/pull/324729
- [ ] mohawk https://github.com/NixOS/nixpkgs/pull/324732
- [ ] mygpoclient https://github.com/NixOS/nixpkgs/pull/324734
- [x] nanotime https://github.com/NixOS/nixpkgs/pull/311042
- [ ] nbmerge https://github.com/NixOS/nixpkgs/pull/324749
- [ ] nose3
- [x] nose-cov <s>https://github.com/NixOS/nixpkgs/pull/313182</s> https://github.com/NixOS/nixpkgs/pull/313476
- [x] nose-cprof https://github.com/NixOS/nixpkgs/pull/313269
- [ ] nose
- [x] nose-exclude https://github.com/NixOS/nixpkgs/pull/313483
- [x] nosejs https://github.com/NixOS/nixpkgs/pull/313163
- [ ] nose-pattern-exclude
- [x] nose-randomly https://github.com/NixOS/nixpkgs/pull/313621
- [ ] nose-timer
- [ ] nose-warnings-filters
- [x] nosexcover https://github.com/NixOS/nixpkgs/pull/313268
- [ ] ofxhome
- [ ] ofxtools
- [x] opuslib https://github.com/NixOS/nixpkgs/pull/311056
- [ ] paramz
- [ ] pid
- [ ] piep https://github.com/NixOS/nixpkgs/issues/311054#issue-2291320318
- [ ] preggy
- [ ] premailer
- [x] proboscis https://github.com/NixOS/nixpkgs/pull/313168
- [ ] ptable
- [x] pxml https://github.com/NixOS/nixpkgs/pull/311201
- [ ] pycdio
- [ ] pycontracts
- [x] pycron https://github.com/NixOS/nixpkgs/pull/313262
- [ ] pygatt
- [ ] pygeoip
- [ ] pynose
- [ ] pypass
- [ ] pypillowfight
- [ ] pyquaternion
- [ ] pysrt
- [ ] python-etcd
- [ ] python-hglib
- [x] pyzufall https://github.com/NixOS/nixpkgs/pull/311571
- [x] rarfile https://github.com/NixOS/nixpkgs/pull/311377
- [ ] rx
- [ ] safe
- [ ] sampledata
- [ ] scales
- [ ] selectors2
- [x] shouldbe https://github.com/NixOS/nixpkgs/pull/311072
- [ ] simplebayes
- [ ] spark-parser
- [x] sqlsoup https://github.com/NixOS/nixpkgs/pull/311370
- [ ] sure
- [ ] svgutils
- [ ] tempita
- [ ] theano https://github.com/NixOS/nixpkgs/pull/313148
- [ ] tissue
- [ ] traittypes
- [ ] trfl
- [ ] uncompyle6
- [ ] vxi11
- [ ] webassets
- [ ] webhelpers
- [x] xlwt https://github.com/NixOS/nixpkgs/pull/318498
- [ ] yanc
- [ ] yapf
- [ ] yarg
- [ ] zipstream

### PR DESCRIPTION
**Pull request by ijaketak**
## Description of changes

Fixes #311054.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/python-hglib/default.nix [hunk 1]
```diff
@@ -1,24 +1,28 @@
...
-  fetchPypi,
+  fetchhg,
   mercurial,
-  nose,
 }:
 
 buildPythonPackage rec {
   pname = "python-hglib";
-  version = "2.6.2";
+  version = "2.6.2+nixpkgs.1";
```
Comment (applePrincess) [inline pkgs/development/python-modules/python-hglib/default.nix:10]: according to https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md

```suggestion
  version = "2.6.2-unstable-2024-06-17";
```

**Review by applePrincess — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/python-hglib/default.nix (hunk 1 above)
Comment (ijaketak) [reply pkgs/development/python-modules/python-hglib/default.nix:10]: Naive application of your suggestion causes:
```
setuptools.extern.packaging.version.InvalidVersion: Invalid version: '2.6.2-unstable-2024-06-17'
```

`setuptools` requires a valid version which `setup.py` can detect.
`.hg_archival.txt` or somewhere should include the valid version string `2.6.2+nixpkgs.1`, but it is different from a version placed in derivation. Is it ok?

Comment (applePrincess) [reply pkgs/development/python-modules/python-hglib/default.nix:10]: I could not find the consensus (e.g. documentation or issues) on how to report _package revision_ but it seems like keep the version as is.

https://github.com/NixOS/nixpkgs/blob/39fb5bf8103700d85643a9733fb4156804a59196/pkgs/development/python-modules/tubes/default.nix#L13


So in this case, remove temporary versioning? Not sure.

Comment (ijaketak) [reply pkgs/development/python-modules/python-hglib/default.nix]: Applied your suggestion.

**Comment by applePrincess** [comment]
Result of `nixpkgs-review pr 325681` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>mozphab</li>
    <li>mozphab.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.powerline</li>
    <li>python311Packages.powerline.dist</li>
    <li>python311Packages.python-hglib</li>
    <li>python311Packages.python-hglib.dist</li>
    <li>python312Packages.python-hglib</li>
    <li>python312Packages.python-hglib.dist</li>
    <li>sourcehut.hgsrht</li>
    <li>sourcehut.hgsrht.dist</li>
  </ul>
</details>

**Comment by applePrincess** [comment]
Looks good to me. two packages failing is due to `python3Packages.glean-sdk` failed to build.
just squash two commits into one

`python3Packages.python-hglib: 2.6.2 -> 2.6.2-unstable-2024-06-17`

**Review by applePrincess — APPROVED** [review]

**Comment by ijaketak** [comment]
Closed in favor of #330739.

**Pull request closed without merging**